### PR TITLE
Make it compatible with eslint v5

### PR DESCRIPTION
**Pull request by ferrannp**
- [x] Update compatible airbnb config
- [x] Update eslint-plugin-jsx-a11y 
- [x] Testing in a local project

## Annoying rules

This ones I silent them in my test project:

```js
"react/destructuring-assignment": 0,
"prefer-destructuring": 0
```

## Issues

This should not be happening:

```js
// eslint issue here: react/no-unused-state
state = {
  index: 0,
  routes: [
    { key: '1', title: I18n.t('available') },
    { key: '2', title: I18n.t('upcoming') },
    { key: '3', title: I18n.t('all') },
  ],
};

// eslint issue here: react/no-unused-state
_handleIndexChange = index => this.setState({ index });

<TabViewAnimated
  navigationState={this.state}
  ...
/>
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-callstack-io",
-  "version": "1.1.2",
+  "version": "2.0.0",
```
Comment (thymikee) [inline package.json]: just don't change the version, we'll do it in a publish step

File: package.json [hunk 2]
```diff
@@ -12,21 +12,21 @@
...
+    "babel-eslint": "^8.2.6",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-config-prettier": "^3.0.1",
+    "eslint-plugin-flowtype": "^2.50.0",
+    "eslint-plugin-import": "^2.14.0",
+    "eslint-plugin-jest": "^21.21.0",
+    "eslint-plugin-jsx-a11y": "^6.1.1",
+    "eslint-plugin-prettier": "^2.6.2",
     "eslint-plugin-react": "^7.5.1",
     "prettier": "^1.10.2"
```
Comment (thymikee) [inline package.json]: bump prettier too? :D

File: README.md [hunk 3]
```diff
@@ -45,3 +45,16 @@ Add to your eslint config (`.eslintrc`, or `eslintConfig` field in `package.json
     "extends": "callstack-io"
 }
 ```
+
+### Example of extending the configuration
+
+```json
+{
+  "extends": "callstack-io",
+    "rules": {
```
Comment (jukben) [inline README.md]: wrong indentation

File: index.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ module.exports = {
   extends: [
     'airbnb',
     'plugin:flowtype/recommended',
+    'plugin:jest/recommended',
```
Comment (jukben) [inline index.js]: 👍 

I guess we don't need whole part

```
  overrides: {
     files: [
       '**/__tests__/**/*.js?(x)',
       '**/__mocks__/**/*.js?(x)',
       '**/?(*.)(spec|test).js?(x)'	
     ],
     env: {	
       'jest/globals': true	 
     }
   },
```

**Pull request merged**